pkg/lib/oauth/handler/handler_token.go

```diff
@@ -707,6 +707,12 @@ func (h *TokenHandler) verifyIDTokenDeviceSecretHash(offlineGrant *oauth.Offline
 	if subtle.ConstantTimeCompare([]byte(offlineGrant.DeviceSecretHash), []byte(deviceSecretHash)) != 1 {
 		err = protocol.NewError("invalid_grant", "the device_secret (actor_token) does not bind to the session")
 	}
+	dpopProof := dpop.GetDPoPProof(ctx)
+	if offlineGrant.DeviceSecretDPoPJKT != "" {
+		if dpopProof == nil || dpopProof.JKT != offlineGrant.DeviceSecretDPoPJKT {
```

Comment:
I suggest we add a method to OfflineGrant to do a constant time compare. In particular it should handle a nil proof.

```golang
dpopProof = dpop.GetDPoPProof(ctx)
if !offlineGrant.CheckDPoPProof(dpopProof) {
  err = ErrInvalidDPoPKeyBinding
}

func (g *OfflineGrant) CheckDPoPProof(dpopProof) bool {
  if g.DeviceSecretDPoPJKT == "" {
    // Not bound; Always valid.
    return true
  }
  if dpopProof == nil {
    // Bound but no proof; Always false.
    return false
  }
  // Do a constant time comparison here.
}
```